### PR TITLE
refactor: upgrade ethers.js v6

### PR DESCRIPTION
**Pull request by neel-ds**
## Description

Upgraded the ethers.js dependency to v6 across most apps. To support this upgrade, the dependency was also upgraded in the libs. Related dependencies like Wagmi, Rainbowkit, and Typescript were updated for compatibility. Currently experimenting with downgrading the typescript version to resolve remaining union-type errors. 

## Related Issue

This PR resolves the issue #366 

## Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

## Other information

- The upgrade was necessary to keep the project up-to-date with the latest ethers.js version and take advantage of new features and bug fixes.
- Downgrading the typescript version is an experimental approach to resolve the remaining union-type errors, which may or may not work. You can find the build error [here](https://gist.github.com/neel-ds/5b26a02f69f5b2d8044f9ae17f6f43c2)
- If the typescript version downgrade doesn't resolve the errors, further investigation and changes may be required.

### CODE REVIEW

**Review by 0xjei — CHANGES_REQUESTED** [review]
Thank you @neel-ds for your patience and effort. I have tried everything locally and the build still fails. By the way, I have managed the next steps to successfully complete the update. You can follow this list.


1. Add `// @ts-ignore` for getting rid of `TS2590: Expression produces a union type that is too complex to represent`. You can add the line before the tags in the line that the build error indicates to you. We can open an issue and tackle this error later since is non-blocking.
2. Update `vite` and `@vitejs/plugin-react` to the latest versions on the `dashboard` app.
3. Downgrade `chai` on contracts package to `4.3.7` since >= v5 is ESM only.
4. Upodate `@typechain/hardhat` and `@nomicfoundation/hardhat-ethers` to latest version on `contracts` app
5. Add the following missing dev dependency on `contracts` app
  - `@nomicfoundation/hardhat-ignition`
  - `@nomicfoundation/hardhat-ignition-ethers`
  - `@nomicfoundation/ignition-core`
6. On the `dashboard` app `index.html` add the following script below the `<body>` tag.
    ```html
    <script>
        var global = global || window
    </script>
    ```
7. Update `siwe` package to latest on `api` and `dashboard` apps.
8. Add missing Node polyfills to `dashaboard` app.
   - Install `vite-plugin-node-polyfills` as dev dependency
   - Add them on `vite.config.ts`
9. Your PR introduces the `PROJECT_ID` environemnt variable in the `auth-context.tsx` file on the `dashboard` app.
   - Rename this to `VITE_PROJECT_ID`
   - Add the variable on the `.env.local`, `.env.production` and `.env.staging` with the value `"bandada"`.
   - use `import.meta.env.VITE_PROJECT_ID!` instead of `process.env.PROJECT_ID!`
10. The `auth_service.ts` file in the `api` app use `siweMessage.validate()`. This is deprecated, you can change it with `siweMessage.verify()` and change the expected output accordingly.

Let me know if you have any more comments / errors / thoughts on this. Happy to chat and gl with the remaining todos (should be quick!).

**Comment by neel-ds** [comment]
Thanks @0xjei, I will do the needful promptly.

**Comment by neel-ds** [comment]
Thank you very much @0xjei for continuous support. I have completed the requested changes and rebased the branch with `dev` branch.

**Comment by 0xjei** [comment]
As we have discussed 

> Thank you very much @0xjei for continuous support. I have completed the requested changes and rebased the branch with `dev` branch.

Thank you so much for your effort @neel-ds. Could you please run `yarn prettier:write && yarn lint --fix` from the root of the project? Also, check if the tests (`yarn test`) are passing since CI is failing! Thank you again!

File: apps/contracts/test/BandadaSemaphore.ts [hunk 1]
```diff
@@ -12,9 +12,9 @@ describe("BandadaSemaphore", () => {
     let bandada: Bandada
     let bandadaSemaphore: BandadaSemaphore
 
-    const groupId = utils.formatBytes32String("Name")
+    const groupId = ethers.encodeBytes32String("Name")
     const identities = [0, 1].map((i) => new Identity(i.toString()))
-    const group = new Group(BigNumber.from(groupId).toBigInt())
+    const group = new Group(BigInt(groupId))
 
     group.addMembers(identities.map(({ commitment }) => commitment))
```
Comment (0xjei) [inline apps/contracts/test/BandadaSemaphore.ts:20]: Change L28 `bandada: bandada.address` with `bandada: bandada.target`

**Review by 0xjei — CHANGES_REQUESTED** [review]
Great @neel-ds, thanks again for the extraordinary progress! Only the last 2 steps are missing eheh
- For the failed test on `auth.service` (`Should throw an error if the signature is invalid`) you can delete this test case completely. After some research, the test is wrong and not needed.
- Use `bandada.target` instead of `bandada.address` as mentioned in the other comment.

Looking forward to your changes :sunglasses:

**Review by 0xjei — APPROVED** [review]
Amazing work @neel-ds!!! Thank you so much for your effort, patience and work! Looking forward to collab again with you :)

**Review by 0xjei — APPROVED** [review]

**Pull request merged**